continuous-integration/readme.md

```diff
@@ -1,3 +1,235 @@
+- Reduce human errors, which is probably the most important part of automating the builds
+
+## Single Source Repository
+
+- [ ] **Code / manifest artifacts required to build your project should be maintained in a single git repository.**
+
+  - CI provider-specific build [pipeline definition(s)](https://docs.microsoft.com/en-us/azure/devops/pipelines/get-started/pipelines-get-started?view=azure-devops#define-pipelines-using-yaml-syntax) should reside within the mainline repository.
+
+## Build Automation
+
+An automated build should encompass the following principles:
+
+- [ ] **Build Task**
+  - A single step within your build pipeline that compiles your code project into a single build artifact.
+- [ ] **Unit Testing**
+  - Your build definition includes validation steps to execute a suite of automated unit tests to ensure that application components meets its design and behaves as intended.
+- [ ] **Code Style Checks**
+  - Code across an engineering team must be formatted to agreed coding standards. Such standards keep code consistent and most importantly easy for the team and customer(s) to read and refactor. Code styling consistency encourages collective ownership for project scrum teams and our partners.
+  - There are several open source code style validation tools available to choose from ([code style checks](https://github.com/checkstyle/checkstyle), [StyleCop](https://en.wikipedia.org/wiki/StyleCop)). The [Code Review section](https://github.com/microsoft/code-with-engineering-playbook/tree/master/code-reviews#language-specific-guidance) of the playbook has suggestions for linters and preferred styles for a number of languages.
+  - Code standards are maintained within a single configuration file. There should be a step in your build pipeline that asserts code in the latest commit conforms to the known style definition.
+- [ ] **Database Schema Automation**
+  - The build process should automate pulling database schemas from the git repository and applied to the target environment. This also includes automating the steps to seed that environment with the necessary test data required to run isolated end-to-end automated tests.
+- [ ] **Build Script Target**
+  - A single command should have the capability of building the system. This is also true for builds running on a CI server or on a developers local machine.
+- [ ] **No IDE Dependencies**
+  - It's essential to have a master build that's runnable through standalone scripts and not dependent on a particular IDE. Build pipeline targets can be triggered locally on their desktops through their IDE of choice. The build process should maintain enough flexibility to run within a CI server as well. As an example, dockerizing your build process offers this level of flexibility as VSCode and IntelliJ supports [docker plugin](https://code.visualstudio.com/docs/containers/overview) extensions.
+
+## Build Environment Dependencies
+
+- [ ] **Automated local environment setup**
```

Comment:
This section sounds like its a good candidate to port into https://github.com/microsoft/code-with-engineering-playbook/pull/248